src/userdb/userdbctl.c

```diff
@@ -1652,24 +1664,12 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case ARG_OUTPUT:
-                        if (isempty(optarg))
-                                arg_output = _OUTPUT_INVALID;
-                        else if (streq(optarg, "classic"))
-                                arg_output = OUTPUT_CLASSIC;
-                        else if (streq(optarg, "friendly"))
-                                arg_output = OUTPUT_FRIENDLY;
-                        else if (streq(optarg, "json"))
-                                arg_output = OUTPUT_JSON;
-                        else if (streq(optarg, "table"))
-                                arg_output = OUTPUT_TABLE;
-                        else if (streq(optarg, "help")) {
-                                puts("classic\n"
-                                     "friendly\n"
-                                     "json\n"
-                                     "table");
-                                return 0;
-                        } else
-                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid --output= mode: %s", optarg);
+                        if (streq(optarg, "help"))
+                                return DUMP_STRING_TABLE(output, Output, _OUTPUT_MAX);
+
+                        arg_output = output_from_string(optarg);
+                        if (arg_output < 0 && !isempty(optarg))
```

Comment:
Hmm... Accepting an empty string like that sound spurious. Maybe better to accept an empty string earlier earlier with an explicit comment:
```c
if (isempty(optarg)) /* comment here */
        arg_output = _OUTPUT_INVALID;
else
        arg_output = output_from_string(optarg);
if (arg_output < 0)
        return log_error_errno(.....);
```